### PR TITLE
Add health checks for ManagedResource

### PR DESCRIPTION
**Pull request by timuthy**
**What this PR does / why we need it**:
This PR adds a convenient function which allows to check the `conditions` of `ManagedResources`.

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: pkg/health/health.go [hunk 1]
```diff
@@ -32,6 +33,31 @@ var (
 	}
 )
 
+// CheckManagedResource checks if the conditions of ManagedResources are healthy.
+// It additionally checks whether at least the `v1alpha1.ResourcesApplied` has been reported.
+func CheckManagedResource(mr v1alpha1.ManagedResource) error {
```
Comment (ialidzhikov) [inline pkg/health/health.go]:
```suggestion
func CheckManagedResource(mr *v1alpha1.ManagedResource) error {
```

Comment (timuthy) [reply pkg/health/health.go]: Why do you want to pass a pointer? `CheckManagedResource` does not modify the object.

Comment (ialidzhikov) [reply pkg/health/health.go]: To avoid a copy. All other check funcs accept pointers. However you can ignore the comment if you don't find it relevant.

Comment (timuthy) [reply pkg/health/health.go]: Okay you are right, for sake of consistency.

**Pull request merged**